pom.xml

```diff
@@ -99,6 +106,65 @@
           <skipLocalStaging>true</skipLocalStaging>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <!-- To match the OSGi version format in the Export-Package specification of the manifest, strip off
+             the 'jre' part of the version name (and anything following it, i.e. SNAPSHOT). -->
+        <executions>
+          <execution>
+            <id>regex-property</id>
+            <goals>
+              <goal>regex-property</goal>
+            </goals>
+            <configuration>
+              <name>osgi.version</name>
+              <value>${project.version}</value>
+              <regex>(.*)-jre.*</regex>
+              <replacement>$1.0</replacement>
+              <failIfNoMatch>true</failIfNoMatch>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <version>3.5.0</version>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Bundle-SymbolicName>com.datastax.oss.driver.shaded.guava</Bundle-SymbolicName>
+            <!-- Export-Package and Import-Package must be declared explicitly since the maven-bundle-plugin executes before maven-shade-plugin, so the guava classes aren't present at this time.
+                 Note that these should match the specifications from the MANIFEST.mf file of the guava jar this jar will be based on. -->
+            <Export-Package>
```

Comment:
This does look kind of hacky, but was best solution I could find.  Also it's pretty easy to extract from the Guava MANIFEST and then do a search and replace.

Reply:
I tried fixing the execution order but it's not enough.

If you use non-default executions for both plugins, they will run in the order they're declared in the POM:

```xml
      <plugin>
        <artifactId>maven-shade-plugin</artifactId>
        <executions>
          <execution>
            <id>relocate-classes</id>
            <goals><goal>shade</goal></goals>
            <configuration> ... </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-bundle-plugin</artifactId>
        <executions>
          <execution>
            <id>write-osgi-manifest</id>
            <goals><goal>bundle</goal></goals>
            <configuration> ... </configuration>
          </execution>
        </executions>
      </plugin>
```
The shade plugin replaces the artifact first:
```
[INFO] Replacing original artifact with shaded artifact.
```
But the bundle plugin still doesn't see it:

```
[WARNING] Bundle com.datastax.oss:java-driver-shaded-guava:bundle:25.1-jre-SNAPSHOT : The JAR is empty: The instructions for the JAR named com.datastax.oss.driver.shaded.guava did not cause any content to be included, this is likely wrong
```

I ran into similar issues while working on shading the driver. It looks like the JAR is replaced at the filesystem level, but other plugins in the reactor are still operating on the pre-shaded structure (maybe held in memory).

Anyway this module shouldn't change often, so it's not worth spending too much time on this.